az-hadoop-jobtype-plugin/src/main/java/azkaban/jobtype/HadoopJobUtils.java

```diff
@@ -433,52 +442,65 @@ public Void run() throws Exception {
    *
    * @return a Set. May be empty, but will never be null
    */
-  public static Set<String> findApplicationIdFromLog(String logFilePath, Logger log) {
-
-    File logFile = new File(logFilePath);
+  public static Set<String> findApplicationIdFromLog(final String logFilePath, final Logger log) {
+    // At least one job log file must be there.
+    final File logFile = new File(logFilePath);
 
     if (!logFile.exists()) {
       throw new IllegalArgumentException("the logFilePath does not exist: " + logFilePath);
     }
-    if (!logFile.isFile()) {
-      throw new IllegalArgumentException("the logFilePath specified  is not a valid file: "
-          + logFilePath);
-    }
-    if (!logFile.canRead()) {
-      throw new IllegalArgumentException(
-          "unable to read the logFilePath specified: " + logFilePath);
-    }
+
+    // Get the directory and file name
+    final int lastSlash = logFilePath.lastIndexOf('/');
+    final String dirPath = logFilePath.substring(0, lastSlash);
+    final String logFileName = logFilePath.substring(lastSlash + 1);
+
+    final File[] logFiles =
+        new File(dirPath).listFiles((dir, name) -> name.startsWith(logFileName));
 
     BufferedReader br = null;
-    Set<String> applicationIds = new HashSet<String>();
+    final Set<String> applicationIds = new HashSet<>();
 
     try {
-//      br = new BufferedReader(new FileReader(logFile));
-      br = new BufferedReader(new InputStreamReader(
-          new FileInputStream(logFile), StandardCharsets.UTF_8));
-      String line;
-
-      // finds all the application IDs
-      while ((line = br.readLine()) != null) {
-        String[] inputs = line.split("\\s");
-        if (inputs != null) {
-          for (String input : inputs) {
-            Matcher m = APPLICATION_ID_PATTERN.matcher(input);
+      // There can be more than one log file. Go through each one of them.
```

Comment:
In what scenarios are there more than one log file?

Reply:
When the log file goes over 5MB. Jobrunner uses RollingAppender for that purpose.
That is why the MR jobs were leaking.
As can be seen in the test, the file names are named according to how they would be in real use case.